crates/router/src/core/subscription.rs

```diff
@@ -63,3 +77,556 @@ pub async fn create_subscription(
+                self.handler
+                    .merchant_context
+                    .get_merchant_account()
+                    .get_id(),
+                &mca_id,
+                self.handler.merchant_context.get_merchant_key_store(),
+            )
+            .await
+            .change_context(errors::ApiErrorResponse::MerchantConnectorAccountNotFound {
+                id: mca_id.get_string_repr().to_string(),
+            })?;
+
+        let connector_name = billing_processor_mca.connector_name.clone();
+
+        let auth_type: hyperswitch_domain_models::router_data::ConnectorAuthType =
+            payments_core::helpers::MerchantConnectorAccountType::DbVal(Box::new(
+                billing_processor_mca.clone(),
+            ))
+            .get_connector_account_details()
+            .parse_value("ConnectorAuthType")
+            .change_context(errors::ApiErrorResponse::InvalidDataFormat {
+                field_name: "connector_account_details".to_string(),
+                expected_format: "auth_type and api_key".to_string(),
+            })?;
+
+        let connector_data = api_types::ConnectorData::get_connector_by_name(
+            &self.handler.state.conf.connectors,
+            &connector_name,
+            api_types::GetToken::Connector,
+            Some(billing_processor_mca.get_id()),
+        )
+        .change_context(errors::ApiErrorResponse::IncorrectConnectorNameGiven)
+        .attach_printable(
+            "invalid connector name received in billing merchant connector account",
+        )?;
+
+        let connector_enum =
+            common_enums::connector_enums::Connector::from_str(connector_name.as_str())
+                .change_context(errors::ApiErrorResponse::InternalServerError)
+                .attach_printable(format!("unable to parse connector name {connector_name:?}"))?;
+
+        let connector_params =
+            hyperswitch_domain_models::connector_endpoints::Connectors::get_connector_params(
+                &self.handler.state.conf.connectors,
+                connector_enum,
+            )
+            .change_context(errors::ApiErrorResponse::ConfigNotFound)
+            .attach_printable(format!(
+                "cannot find connector params for this connector {connector_name} in this flow",
+            ))?;
+
+        Ok(BillingHandler {
+            subscription: self.subscription.clone(),
+            connector_name,
+            auth_type,
+            connector_data,
+            connector_params,
+            request: self.handler.request.clone(),
+            connector_metadata: billing_processor_mca.metadata.clone(),
+        })
+    }
+
+    pub async fn get_invoice_handler(&self) -> errors::RouterResult<InvoiceHandler> {
+        Ok(InvoiceHandler {
+            subscription: self.subscription.clone(),
+        })
+    }
+}
+
+pub struct BillingHandler {
+    subscription: diesel_models::subscription::Subscription,
+    connector_name: String,
```

Comment:
Connector_name is already available in connector_data, this field is not necessary.